File: app/schemas/goal.py
"""Pydantic schemas for daily/weekly learning goals."""

from datetime import datetime
from pydantic import BaseModel, Field


class GoalCreate(BaseModel):
    goal_type: str = Field(..., pattern="^(daily|weekly)$")
    description: str = Field(..., min_length=1, max_length=500)
    target_value: int | None = None
    unit: str = "minutes"
    target_date: str  # YYYY-MM-DD


class GoalUpdate(BaseModel):
    progress: int | None = None
    completed: int | None = None


class GoalResponse(BaseModel):
    id: int
    goal_type: str
    description: str
    target_value: int | None = None
    progress: int
    unit: str
    target_date: str
    completed: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/goal.ts
/** Goal types — mirrors app/schemas/goal.py */

export type GoalType = "daily" | "weekly";

export interface Goal {
  id: number;
  goal_type: GoalType;
  description: string;
  target_value: number | null;
  progress: number;
  unit: string;
  target_date: string;
  completed: number; // 0 or 1
  created_at: string;
}

export interface GoalCreate {
  goal_type: GoalType;
  description: string;
  target_value?: number | null;
  unit?: string;
  target_date: string; // YYYY-MM-DD
}

export interface GoalUpdate {
  progress?: number;
  completed?: number;
}

export interface TodayGoalsResponse {
  goals: Goal[];
  suggestions: GoalSuggestion[];
}

export interface GoalSuggestion {
  description: string;
  target_value: number;
  unit: string;
}
